### PR TITLE
Upgrade Gradle profiler to latest alpha version

### PR DESCRIPTION
**Pull request by wolfs**
So we can capture multiple events with Gradle profiler, e.g. allocations and CPU at the same time.

This improves the layout of the files on disk as well. After this change, the output directory per test will look like this for cross version tests:
image

And like this for cross-build tests:
image

Do not merge, yet, since we need to do newer release of Gradle profiler to contain the latest changes from there.

### CODE REVIEW

File: subprojects/build-scan-performance/src/testFixtures/groovy/org/gradle/performance/AbstractBuildScanPluginPerformanceTest.groovy [hunk 1]
```diff
@@ -28,16 +28,12 @@ import org.gradle.performance.measure.MeasuredOperation
...
 import org.gradle.performance.results.CrossBuildPerformanceResults
-import org.gradle.performance.results.GradleProfilerReporter
-import org.gradle.test.fixtures.file.CleanupTestDirectory
 import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
 import org.junit.Rule
 import spock.lang.AutoCleanup
 import spock.lang.Shared
-import spock.lang.Specification
 
-@CleanupTestDirectory
```
Comment (blindpirate) [inline subprojects/build-scan-performance/src/testFixtures/groovy/org/gradle/performance/AbstractBuildScanPluginPerformanceTest.groovy:39]: Why this is removed?

Comment (blindpirate) [reply subprojects/build-scan-performance/src/testFixtures/groovy/org/gradle/performance/AbstractBuildScanPluginPerformanceTest.groovy:39]: Seems like we're not using `temporaryFolder.testDirectory` any more, does gradle-profiler clean it? What if we keep this annotation, I assume it does nothing in this case?

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/CrossVersionPerformanceTestRunner.groovy [hunk 2]
```diff
@@ -277,6 +277,7 @@ class CrossVersionPerformanceTestRunner extends PerformanceTestSpec {
             .warmUpCount(warmUpRuns)
             .invocationCount(runs)
             .buildMutators(buildMutators)
+            .multiVersion(true)
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/CrossVersionPerformanceTestRunner.groovy:280]: What does `multiVersion` mean here? Does it mean `crossVersion`? If so why do we change it to another terminology?

**Review by blindpirate — COMMENTED** [review]
Awesome! We finally get rid of diff/flamegraph tools in Gradle! I have some questions.

File: subprojects/build-scan-performance/src/testFixtures/groovy/org/gradle/performance/AbstractBuildScanPluginPerformanceTest.groovy (hunk 1 above)
Comment (wolfs) [reply subprojects/build-scan-performance/src/testFixtures/groovy/org/gradle/performance/AbstractBuildScanPluginPerformanceTest.groovy:39]: I added the annotation to the superclass. That should work, right? TemporaryFolder.testDirectory is always removed, the annotation causes to not remove it if the test fails, right?

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/CrossVersionPerformanceTestRunner.groovy (hunk 2 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/CrossVersionPerformanceTestRunner.groovy:280]: Yeah, you are right. I should rename the method to `crossVersion`.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/DefaultOutputDirSelector.groovy [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+package org.gradle.performance.results
+
+import com.google.common.base.Strings
+import groovy.transform.CompileStatic
+
+@CompileStatic
+class DefaultOutputDirSelector implements OutputDirSelector {
+    private static final String DEBUG_ARTIFACTS_DIRECTORY_PROPERTY_NAME = "org.gradle.performance.debugArtifactsDirectory"
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/DefaultOutputDirSelector.groovy:24]: I know this property is from legacy code, but do you see anywhere we set it? I searched a little bit but didn't find anywhere to set it. If nobody is setting it I think we can totally remove this one.

**Review by blindpirate — COMMENTED** [review]
I have only one comment regarding the magic property, others look good to me.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/DefaultOutputDirSelector.groovy (hunk 3 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/DefaultOutputDirSelector.groovy:24]: Yes, we still use this property, and it is important: https://github.com/gradle/gradle/blob/ed9d43311b646d05bbd65a88db1a38b8e015fa30/build-logic/performance-testing/src/main/groovy/gradlebuild/performance/tasks/PerformanceTest.groovy#L320

**Review by blindpirate — APPROVED** [review]
OK, then LGTM!

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**